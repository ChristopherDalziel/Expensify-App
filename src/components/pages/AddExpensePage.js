import React from "react";
import { connect } from "react-redux";
import styled from "@emotion/styled";
import ExpenseForm from "../ExpenseForm";
import { startAddExpense } from "../../actions/expenses";

const AddExpensePageContainer = styled.div`
  margin: 10px;
`;

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    // The addExpense function we set up earlier takes an object with all of the values that the expense prop we're passing up is holding.
    this.props.startAddExpense(expense);
    this.props.history.push("/");
  };
  render() {
    return (
      <AddExpensePageContainer>
        <h1>Add Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </AddExpensePageContainer>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    startAddExpense: (expense) => dispatch(startAddExpense(expense)),
  };
};

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
