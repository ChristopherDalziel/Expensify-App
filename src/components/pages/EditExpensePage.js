import React from "react";

const EditExpensePage = (props) => {
  return (
    <>
      <h1>Edit Expense</h1>
      <p>Editing expense {props.match.params.id}</p>
    </>
  );
};

export default EditExpensePage;
