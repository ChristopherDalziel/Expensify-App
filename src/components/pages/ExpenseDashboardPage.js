import React from "react";

const ExpenseDashboardPage = () => {
  return <div>Dashboard Page</div>;
};

export default ExpenseDashboardPage;
