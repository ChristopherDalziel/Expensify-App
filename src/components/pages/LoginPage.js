import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../../actions/auth";
import Button from "react-bootstrap/Button";
import styled from "@emotion/styled";

const LoginContainer = styled.main`
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const LoginPage = ({ startLogin }) => {
  return (
    <LoginContainer>
      <Button variant="outline-primary" onClick={startLogin}>
        Login
      </Button>
    </LoginContainer>
  );
};

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
