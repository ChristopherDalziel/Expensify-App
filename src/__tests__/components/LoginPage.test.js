import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { LoginPage } from "../../components/pages/LoginPage";

test("Should render LoginPage correctly", () => {
  const wrapper = shallow(<LoginPage />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("Should call startLogin on button click", () => {
  const startLoginSpy = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLoginSpy} />);
  wrapper.find("Button").simulate("click");
  expect(startLoginSpy).toHaveBeenCalledTimes(1);
});
