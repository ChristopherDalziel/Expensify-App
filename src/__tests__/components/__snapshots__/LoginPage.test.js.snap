// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage correctly 1`] = `
<div>
  <Button
    active={false}
    disabled={false}
    type="button"
    variant="outline-primary"
  >
    Login
  </Button>
</div>
`;
